src/main/java/org/folio/rest/impl/PatronServicesResourceImpl.java

```diff
@@ -214,7 +211,22 @@ private CompletableFuture<Response> handleUserUpdateResponse(JsonObject userResp
             .thenCompose(addressTypes -> {
               if (Objects.equals(patronGroup, remotePatronGroupId)) {
               final String homeAddressTypeId = addressTypes.getString(HOME_ADDRESS_TYPE);
-                return updateUser(userId, entity, okapiHeaders, userRepository, remotePatronGroupId, homeAddressTypeId);
+                final String entityEmail = entity.getContactInfo().getEmail();
+                if (!emailId.equals(entityEmail)) {
+                  return getUserByEmail(entityEmail, okapiHeaders, userRepository)
+                    .thenCompose(userEmailResponse -> {
+                      final int records = userEmailResponse.getInteger(TOTAL_RECORDS);
+                      if (records > 0) {
+                        logger.error("putPatronAccountByEmailByEmailId:: User already exist with email provided in payload");
+                        return CompletableFuture.completedFuture(
+                          PutPatronAccountByEmailByEmailIdResponse.respond500WithTextPlain("User already exist with email provided in payload"));
```

Comment:
I think response code should be 4XX instead of 5XX.

Reply:
fixed